### PR TITLE
[WIP] Add datasets from "Encoding high-cardinality string categorical variables" paper

### PR DESCRIPTION
**Pull request by TwsThomas**
Adding the 24 datasets from "[Encoding high-cardinality string categorical variables](https://arxiv.org/pdf/1907.01860.pdf)" paper.

Datasets active on OpenML:
- [ ] building_permits
- [x] beer_reviews
- [x] colleges
- [x] crime_data (dead link :/)
- [ ] drug_directory
- [x] employe_salaries
- [x] federal_election
- [x] journal_influence
- [x] kickstarter_projects
- [x] medical_charge
- [ ] met_objects
- [ ] midwest_survey
- [ ] open_payment (dead link :/)
- [x] public_procurement
- [ ] road_safety (dead link :/)
- [x] traffic_violations
- [x] vancouver_employee
- [x] wine_reviews
- [x] Adult (from UCI, https://www.openml.org/d/1590)
- [x] cacao_flavor
- [x] california_housing (https://www.openml.org/d/537)
- [x] dating_profile
- [x] house_prices
- [x] house_sales
- [x] intrusion_detection (https://www.openml.org/d/1113)

I've drop few rows (in colleges and la_crime) when label was missing. (It seems it was required from openml).
In public_procurement, I replace nan by -1 in the label `award_value_euro`

### CODE REVIEW

**Comment by GaelVaroquaux** [comment]
Would the code here be simplified once those datasets make it to openML?

**Comment by TwsThomas** [comment]
> Would the code here be simplified once those datasets make it to openML?

Yes!
This code is just to record the upload process. 
Once in openML, this will be a one line like:
`sklearn.datasets.fetch_openml(data_id=xxx)`

**Comment by GaelVaroquaux** [comment]
> This code is just to record the upload process.

Of course! Great, super useful!!

**Comment by GaelVaroquaux** [comment]
Should we merge this with what you have, and we open another PR?

**Comment by TwsThomas** [comment]
> Should we merge this with what you have, and we open another PR?

Yes. I'll open a new one to upload the last datasets.
